django/core/management/commands/shell.py

```diff
@@ -111,9 +114,35 @@ def python(self, options):
         # Start the interactive interpreter.
         code.interact(local=imported_objects)
 
+    def get_apps_and_models(self):
+        apps_and_models = {}
+        for app_conf in apps.get_app_configs():
+            if app_conf.models_module:
+                apps_and_models[app_conf.models_module] = app_conf.get_models()
+        return apps_and_models
+
+    def get_namespace(self, options):
+        apps_and_models = self.get_apps_and_models()
+        imported_objects = {}
+        for app_module, app_models in reversed(apps_and_models.items()):
+            if not app_models:
+                continue
+            for model in app_models:
+                if model.__module__:
+                    imported_objects[model.__name__] = model
+        if not options["command"]:
+            self.stdout.write(
+                f"{len(imported_objects)} objects imported automatically",
+                self.style.SUCCESS,
+            )
```

Comment:
We can’t have the message inside this method if it’s going to be the one users will override, because the message will then be displayed *before* custom imports are added. I think we will need a wrapper method that runs `get_namespace()` and displays the message.